File: classify/two-pointers/remove-element.c
int removeElement(int* nums, int numsSize, int val) {
    int count = -1;
    int i;

    for(i = 0; i < numsSize; ++i)
        if(nums[i] != val)
            nums[++count] = nums[i];

    return count+1;
}

File: go/array/remove-element.go
func removeElement(nums []int, val int) int {
    i := 0
    j := len(nums) - 1
    for i <= j {
        for i <= j && nums[i] != val {
            i++
        }

        for j >= i && nums[j] == val {
            j--
        }

        if(i < j) {
            nums[i], nums[j] = nums[j], nums[i]
        }
    }

    return i
}
